src/components/overview-table.js

```diff
@@ -131,6 +127,7 @@ function FileDataRow({ processFile, availableInputs, fileGroup, timestamp }) {
 
     const handleClose = () => {
         setOpen(false);
+        setSelectedFilename('');
     };
 
     const selectFile = useCallback(
```

Comment:
As discussed, the `selectedFilename` must be a State variable to work as expected. But I guess the `selectFile` function could be a standard function and not a Callback.
I tried something like this on my own and it seemed to work properly:
```java
    const selectFile = async () => {
        await fetchTaskManagerSelectFile(timestamp, fileType, selectedFilename);
        handleClose();
    };
```